src/meta-srv/src/handler/failure_handler/runner.rs

```diff
@@ -121,16 +128,26 @@ impl FailureDetectRunner {
         self.receiver_handle = Some(receiver_handle);
 
         let election = self.election.clone();
+        let region_failover_manager = self.region_failover_manager.clone();
         let runner_handle = common_runtime::spawn_bg(async move {
             loop {
                 let start = Instant::now();
 
                 let is_leader = election.as_ref().map(|x| x.is_leader()).unwrap_or(true);
                 if is_leader {
-                    for e in failure_detectors.iter() {
-                        if e.failure_detector().is_available(current_time_millis()) {
-                            // TODO(LFC): TBC
-                        }
+                    let failed_regions = failure_detectors
+                        .iter()
+                        .filter_map(|e| {
+                            if !e.failure_detector().is_available(current_time_millis()) {
```

Comment:
Move `current_time_millis()` out of the iterator

```rust
let now = current_time_millis();
 let failed_regions = failure_detectors
                        .iter()
                        .filter_map(|e| {
                            if !e.failure_detector().is_available(now) {
       ....
```

Reply:
I intentionally not place `current_time_millis()` out of the iteration, because I think failure detection determination should be happened "just in time", i.e., failed or not has to be compared with the most recently "now". Besides, it might reduce the false positive of failure detection, because during the iteration, heartbeats are coming in as usual, and the `phi`s are still updating.